### PR TITLE
[release/v1.4] Add missing calico logs volumeMount

### PR DESCRIPTION
**Pull request by doflamingo721**
There was a missing volumeMount which was causing the calico/canal logs getting written to the container overlay filesystem causing the container to fill up. Adding the missing bind resolves this issue.

Linked issue: https://github.com/rancher/rancher/issues/40330
SURE: 5814

### CODE REVIEW

**Review by snasovich — APPROVED** [review]
LGTM on the surface, please add to the description on the PR that it also brings changes from https://github.com/rancher/kontainer-driver-metadata/pull/1076 to RKE.

Still need @kinarashah's approval before merging.

**Comment by doflamingo721** [comment]
@kinarashah As discussed offline, I have added the k8s version constraints so that a existing cluster with a older k8s version than the latest patch won't be auto-upgraded as per the new plan after adding a worker node. Can you please review these changes?

**Comment by doflamingo721** [comment]
Rebased and resolved conflicts. Ready for review. 
@snasovich @kinarashah

**Review by kinarashah — CHANGES_REQUESTED** [review]
All versions need to update to latest patch getting released with April patch instead of Jan to avoid out of band updates of node plan.

**Comment by doflamingo721** [comment]
Since this fix is targeted for the april patches, I have raised another PR to add this fix in the `release/v1.4.5` branch in RKE. Closing this PR in favour of https://github.com/rancher/rke/pull/3218

**Pull request closed without merging**
